elf_reader.go

```diff
@@ -570,10 +575,17 @@ func (ec *elfCode) loadBTFMaps(maps map[string]*MapSpec) error {
 			maps[name] = mapSpec
 		}
 	}
-
 	return nil
 }
 
+func nameFromBTFVarSecinfo(vs btf.VarSecinfo) (string, error) {
+	ss := strings.Split(vs.Type.String(), "\"")
```

Comment:
We should probably consider adding `(btf.Type).Name()` or similar instead, I don't think this is an acceptable approach. :slightly_smiling_face: WDYT @lmb ?

Reply:
> I don't think this is an acceptable approach

I agree

Reply:
The problem is that not all btf Types have a name. There is `namedType interface` in types.go already, what if we expose that?

```
namedType, ok := vs.Type.(btf.NamedType)
if !ok { return err }
namedType.Name()
```

Reply:
I'll give it a try, thanks

Reply:
Seems to work, I called it `GetName` because `Name` clashes with the `Name` type, let me know if I should rename `Name` (the type) in order to have a `Name()` method

Reply:
Sorry, I sent you on a bit of a wild goose chase. Please take a look at: https://github.com/cilium/ebpf/pull/264

Basically, VarSecinfo.Type is guaranteed to be *Var so we don't need all this complication.

Reply:
Thanks @lmb and @ti-mo , I will close this PR for now, and reopen a new one after https://github.com/cilium/ebpf/pull/264 has been merged